### PR TITLE
Tcp udp sg rules tests ok

### PR DESCRIPTION
**Pull request by outscale-toa**

### CODE REVIEW

**Comment by outscale-jju** [comment]
small fixed -> small fixes

File: tests/functions.sh [hunk 1]
```diff
@@ -569,7 +569,7 @@ function delete_rule_port_open {
     nic_id=$3
     protocol=$4
     port=$5
-    echo "update sg rules with only port $port opened in butterfly $but_id"
+    echo "delete sg rules with only port $port opened in butterfly $but_id"
```
Comment (outscale-jju) [inline tests/functions.sh]: "port $port" -> "$protocol port $port"

File: tests/scenario_18/test.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ssh_connection_test udp 1 2 1234
+ssh_connection_test tcp 1 2 1354
+delete_nic 0 2
+delete_nic 0 1
+qemu_stop 1
+qemu_stop 2
+add_nic_void 0 1 42
+add_nic_void 0 2 42
+qemu_start 1
+qemu_start 2
```
Comment (outscale-jju) [inline tests/scenario_18/test.sh]: Replace vm restarting by SG removal

File: tests/functions.sh [hunk 3]
```diff
@@ -365,14 +365,21 @@ function delete_nic {
...
+	tcp_udp_protocol=6
+    elif [ "$1" == "udp" ]; then
+	tcp_udp_protocol=17
+    else
+	echo -e "protocol not set, check protocol please"
+	RETURN_CODE=1
+    fi
     f=/tmp/butterfly-client.req
-    echo "add nic $nic_id with only port $port opened in butterfly $but_id in vni $vni"
+    echo "add nic $nic_id $1 port $port_tcp_udp opened in butterfly $but_id in vni $vni"
```
Comment (outscale-jju) [inline tests/functions.sh]: why don't you set `protocol=$1` like other arguments ?

File: tests/functions.sh [hunk 4]
```diff
@@ -445,14 +445,22 @@ function delete_nic {
...
-    nic_id=$3
-    vni=$4
-    protocol=$5
-    port=$6
+    protocol=$1
+    sg=$2
+    but_id=$3
+    nic_id=$4
+    vni=$5
+    port_tcp_udp=$6
```
Comment (outscale-mgo) [inline tests/functions.sh]: I would prefer if you let port as name

File: tests/functions.sh [hunk 5]
```diff
@@ -644,12 +659,20 @@ function delete_rule_full_open {
...
-    but_id=$2
-    nic_id=$3
-    protocol=$4
-    port=$5
-    echo "update sg rules with only port $port opened in butterfly $but_id"
+    protocol=$1
+    sg=$2
+    but_id=$3
+    nic_id=$4
+    port_tcp_udp=$5
```
Comment (outscale-mgo) [inline tests/functions.sh]: here too, can you just let "port"

File: tests/functions.sh [hunk 6]
```diff
@@ -460,14 +460,22 @@ function delete_nic {
...
-    vni=$4
-    protocol=$5
+    protocol=$1
+    sg=$2
+    but_id=$3
+    nic_id=$4
+    vni=$5
     port=$6
+    if [ "$protocol" == "tcp" ]; then
+	tcp_udp_protocol=6
```
Comment (outscale-mgo) [inline tests/functions.sh]: just reuse $protocol please

File: tests/functions.sh [hunk 7]
```diff
@@ -460,14 +460,22 @@ function delete_nic {
...
+    elif [ "$protocol" == "udp" ]; then
+	tcp_udp_protocol=17
```
Comment (outscale-mgo) [inline tests/functions.sh]: same here

File: tests/functions.sh [hunk 8]
```diff
@@ -460,14 +460,22 @@ function delete_nic {
...
+    else
+	echo -e "protocol not set, check protocol please"
```
Comment (outscale-mgo) [inline tests/functions.sh]: `echo -e "protocol $protocol not supported by add_nic_port_open"`

File: tests/functions.sh [hunk 9]
```diff
@@ -492,7 +500,7 @@ messages {
         id: \"$sg\"
         rule {
           direction: INBOUND
-          protocol: $protocol
+          protocol: $tcp_udp_protocol
```
Comment (outscale-mgo) [inline tests/functions.sh]: so here you can let protocol

File: tests/functions.sh [hunk 10]
```diff
@@ -553,13 +561,21 @@ function update_sg_rule_full_open {
...
-    protocol=$4
+function add_sg_rule_port_open {
+    protocol=$1
+    sg=$2
+    but_id=$3
+    nic_id=$4
     port=$5
-    echo "update sg rules with only port $port opened in butterfly $but_id"
+    if [ "$protocol" == "tcp" ]; then
+	tcp_udp_protocol=6
```
Comment (outscale-mgo) [inline tests/functions.sh]: s/tcp_udp_protocol/protocol/

File: tests/functions.sh [hunk 11]
```diff
@@ -553,13 +561,21 @@ function update_sg_rule_full_open {
...
+    elif [ "$protocol" == "udp" ]; then
+	tcp_udp_protocol=17
```
Comment (outscale-mgo) [inline tests/functions.sh]: s/tcp_udp_protocol/protocol/

File: tests/functions.sh [hunk 12]
```diff
@@ -570,7 +586,7 @@ function update_sg_rule_port_open {
         sg_id: \"$sg\"
         rule {
           direction: INBOUND
-          protocol: $protocol
+          protocol: $tcp_udp_protocol
```
Comment (outscale-mgo) [inline tests/functions.sh]: s/tcp_udp_protocol/protocol/

File: tests/functions.sh [hunk 13]
```diff
@@ -659,12 +674,20 @@ function delete_rule_full_open {
...
-    nic_id=$3
-    protocol=$4
+    protocol=$1
+    sg=$2
+    but_id=$3
+    nic_id=$4
     port=$5
-    echo "update sg rules with only port $port opened in butterfly $but_id"
+    if [ "$protocol" == "tcp" ]; then
+	tcp_udp_protocol=6
```
Comment (outscale-mgo) [inline tests/functions.sh]: s/tcp_udp_protocol/protocol/

File: tests/functions.sh [hunk 14]
```diff
@@ -659,12 +674,20 @@ function delete_rule_full_open {
...
+    elif [ "$protocol" == "udp" ]; then
+	tcp_udp_protocol=17
```
Comment (outscale-mgo) [inline tests/functions.sh]: s/tcp_udp_protocol/protocol/

File: tests/functions.sh [hunk 15]
```diff
@@ -675,7 +698,7 @@ function delete_rule_port_open {
         sg_id: \"$sg\"
         rule {
           direction: INBOUND
-          protocol: $protocol
+          protocol: $tcp_udp_protocol
```
Comment (outscale-mgo) [inline tests/functions.sh]: s/tcp_udp_protocol/protocol/

**Pull request merged**